src/Physics/GenericHydro.cc

```diff
@@ -162,6 +164,14 @@ dt(const DataBase<Dimension>& dataBase,
   // Initialize the return value to some impossibly high value.
   auto minDt = make_pair(std::numeric_limits<double>::max(), string());
 
+  // Define a function for computing the velocity divergence (different for curvilinear coordinates)
+  std::function<double(const Tensor&, const Vector&, const Vector&)> Fdiv = [](const Tensor& DvDxi, const Vector& posi, const Vector& veli) { return DvDxi.Trace(); };
```

Comment:
I'm usually all for type verbosity, but we can make this `auto` if you'd like.

Reply:
I'd like to, but the compiler can't seem to figure it out if I do.

Reply:
This looks like our last issue -- I've tried auto (first thing I tried when I first wrote this in fact), but it doesn't compile.   I think the problem is that I want to conditionally assign to this variable in the succeeding if tests, but the type it resolves to with auto isn't allowing that.  This awful std::function declaration gives me an assignable type I can use.  Any suggestions to improve it?

Reply:
Following suggestions from @mdavis36 , this can now be expressed as an auto statement, so the latest commit resolves this request.